Content.Server/Chemistry/Metabolism/DefaultDrink.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using Content.Server.GameObjects.Components.Nutrition;
+using Content.Shared.Interfaces.Chemistry;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Serialization;
+using Robust.Shared.Serialization;
+
+namespace Content.Server.Chemistry.Metabolism
+{
+    /// <summary>
+    /// Default metabolism for drink reagents. Attempts to find a ThirstComponent on the target,
+    /// and to update it's thirst values.
+    /// </summary>
+    class DefaultDrink : IMetabolizable
+    {
+        //Rate of metabolism in units / second
+        private int _metabolismRate;
+        public int MetabolismRate => _metabolismRate;
+
+        //How much thirst is satiated when 1u of the reagent is metabolized
+        private float _hydrationFactor;
+        public float HydrationFactor => _hydrationFactor;
+
+        void IExposeData.ExposeData(ObjectSerializer serializer)
+        {
+            serializer.DataField(ref _metabolismRate, "rate", 1);
+            serializer.DataField(ref _hydrationFactor, "nutrimentFactor", 30.0f);
+        }
+
+        //Remove reagent at set rate, satiate thirst if a ThirstComponent can be found
+        int IMetabolizable.Metabolize(IEntity solutionEntity, string reagentId, float frameTime)
+        {
+            int metabolismAmount = (int)Math.Round(MetabolismRate * frameTime);
```

Comment:
I'm not fond of this method of handling metabolism. It means that due to rounding errors there could be significant changes in the metabolisation rate of reagents, depending on the exact frame rate being processed at.

Reply:
I messed up the documentation on this function. In this case `frameTime` isn't really the time since the last frame, it's really the time since the last metabolism tick. If you look in [StomachSystem.cs](https://github.com/space-wizards/space-station-14/blob/1f177a044dfbff349af29212682c29925f263f46/Content.Server/GameObjects/EntitySystems/StomachSystem.cs#L21) it only updates `StomachComponent`s once every second. In my tests this is always 1±0.01 seconds. So as long as the server isn't struggling to keep up, it should stay at a reliable 1 update per second. I renamed it to `tickTime` and fixed the documentation so it's not misleading.

I think that eliminates that issue, but if I'm missing your point lemme know.

Reply:
Ah, in that case it's probably fine yeah.